### DISCUSSION TITLE
Passing through a connection parameter when using the Auth0 provider

### DISCUSSION THREAD

**Discussion by ericbondinshur** [nextauthjs/next-auth#6377] [Help] [answered]
I am looking to setup next-auth using an Auth0 provider that then uses a different connection for its authentication, is this possible with next-auth currently? 

![image](https://user-images.githubusercontent.com/118747440/212085696-d1cceb9d-7dc3-420d-9d9e-70aca803e065.png)

We have something similar setup with another application that doesn't use Next and it utilises this library: https://github.com/auth0/auth0-spa-js/ and can pass through a connection param, documented here: https://github.com/auth0/auth0-spa-js/blob/master/src/global.ts#L78

Apologies if this has been extensively covered, I have tried to hunt around but couldn't find anything relevant.

**Comment by tcarterBAMF** [comment, marked as answer]
Hey @ericbondinshur

I just ran across this same issue and found that you can set the desired Auth0 connection by overriding the OAuth authorization params when configuring the Auth0Provider:

```javascript
Auth0Provider.default({
    clientId: '...',
    clientSecret: '...',
    issuer: '...',
    authorization: {
      params: {
        connection: 'google-oauth2',
    },
  },
})
```

**Reply by ericbondinshur** [reply]
Very helpful! Thanks

**Reply by yhavinga** [reply]
Yes also useful to configure e.g. forcing login (instead of silently logging in)
```javascript
Auth0Provider({
  clientId: '...',
  clientSecret: '...',
  issuer: '...',
  authorization: {
    params: {
      prompt: 'login',
    },
  }
}),
```
